tests/python/frontend/onnx/test_forward.py

```diff
@@ -1867,6 +1867,23 @@ def verify_unary_ops(op, x, rtol=1e-5, atol=1e-5):
     verify_unary_ops("Sigmoid", x)
     verify_unary_ops("Softsign", x)
 
+    def verify_unary_ops_fp16(op, x, rtol=1e-5, atol=1e-5):
+        z = helper.make_node(op, ["in1"], ["out"])
+        graph = helper.make_graph(
+            [z],
+            "_test",
+            inputs=[
+                helper.make_tensor_value_info("in1", TensorProto.FLOAT16, list(in_shape)),
+            ],
+            outputs=[helper.make_tensor_value_info("out", TensorProto.FLOAT16, list(out_shape))],
+        )
+        model = helper.make_model(graph, producer_name="_test")
+        verify_with_ort_with_inputs(model, [x], rtol=rtol, atol=atol)
+
+    dtype = "float16"
+    x = np.random.uniform(size=in_shape).astype(dtype)
+    verify_unary_ops_fp16("Reciprocal", x)
+
```

Comment:
I think you could pretty cleanly fuse this into `verify_unary_ops` by adding an optional `dtype` argument. You can use `mapping.NP_TYPE_TO_TENSOR_TYPE[np.dtype(dtype)]` to convert that dtype string into the proper ONNX proto type.